Makefile

```diff
@@ -1,27 +1,102 @@
+# Build targets 
+###############################################################################
+## Builds the controller binary and docker image.
 docker-image: image.created
+image.created: dist/kube-policy-controller
+	# Build the docker image for the policy controller.
+	docker build -t $(CONTAINER_NAME) .
+	touch image.created
 
-# Run the unit tests.
-ut: update-version
-	docker run --rm -v `pwd`:/code \
-	calico/test \
-	nosetests tests/unit -c nose.cfg
+dist/kube-policy-controller:
+	$(MAKE) OS=linux ARCH=amd64 binary-containerized
+	mv dist/kube-policy-controller-linux-amd64 dist/kube-policy-controller
+
+## Populates the vendor directory.
+vendor: glide.yaml
+	# Ensure that the glide cache directory exists.
+	mkdir -p $(HOME)/.glide
+
+	# To build without Docker just run "glide install -strip-vendor"
+	if [ "$(LIBCALICOGO_PATH)" != "none" ]; then \
+          EXTRA_DOCKER_BIND="-v $(LIBCALICOGO_PATH):/go/src/github.com/projectcalico/libcalico-go:ro"; \
+	fi; \
+
+	docker run --rm \
+		-v $(CURDIR):/go/src/$(PACKAGE_NAME):rw $$EXTRA_DOCKER_BIND \
+		-v $(HOME)/.glide:/home/user/.glide:rw \
+		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+		$(CALICO_BUILD) \
+		/bin/sh -c 'cd /go/src/$(PACKAGE_NAME) && glide install -strip-vendor'
+
+ut: vendor
+	./run-uts
+
+# Build the controller binary.
+binary: vendor
+	# Don't try to "install" the intermediate build files (.a .o) when not on linux
+	# since there are no write permissions for them in our linux build container.
+	if [ "$(OS)" == "linux" ]; then \
+		INSTALL_FLAG=" -i "; \
+	fi; \
+	GOOS=$(OS) GOARCH=$(ARCH) CGO_ENABLED=0 go build -v $$INSTALL_FLAG -o dist/kube-policy-controller-$(OS)-$(ARCH) \
+	-ldflags "-X main.VERSION=$(GIT_VERSION)" ./main.go
+
+## Build the controller binary in a container.
+binary-containerized: vendor
+	mkdir -p dist
+	-mkdir -p .go-pkg-cache
+	docker run --rm \
+	  -e OS=$(OS) -e ARCH=$(ARCH) \
+	  -v $(CURDIR):/go/src/$(PACKAGE_NAME):ro \
+	  -v $(CURDIR)/dist:/go/src/$(PACKAGE_NAME)/dist \
+	  -e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+	  -v $(CURDIR)/.go-pkg-cache:/go/pkg/:rw \
+	  $(CALICO_BUILD) sh -c '\
+	    cd /go/src/$(PACKAGE_NAME) && \
+	    make OS=$(OS) ARCH=$(ARCH) binary'
+
+###############################################################################
+# Test targets 
+###############################################################################
+
+## Runs all tests - good for CI. 
+ci: clean docker-image check-copyright ut-containerized st 
+
+## Run the tests in a container. Useful for CI, Mac dev.
+ut-containerized: vendor 
+	-mkdir -p .go-pkg-cache
+	docker run --rm --privileged --net=host \
+		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+		-v $(CURDIR)/.go-pkg-cache:/go/pkg/:rw \
+		-v $(CURDIR):/go/src/$(PACKAGE_NAME):rw \
+		$(CALICO_BUILD) sh -c 'cd /go/src/$(PACKAGE_NAME) && make WHAT=$(WHAT) SKIP=$(SKIP) ut'
 
-# Run system tests.
+GET_CONTAINER_IP := docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}'
+K8S_VERSION=1.7.4
```

Comment:
BTW, k8s 1.7.5 is out. Wish hyperkube had a `latest` or a `dev` tag so we can keep testing the latest k8s versions

Reply:
Yeah.  For these sorts of tests the patch releases don't matter much, but the minor releases are more important.